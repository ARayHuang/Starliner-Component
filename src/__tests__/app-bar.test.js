import React from 'react';
import { shallow, mount } from 'enzyme';
import AppBar from '../components/app-bar';

describe('app bar', () => {
	it('should render correctly', () => {
		const props = {
			className: 'mock-app-bar-className',
			position: AppBar.PositionType.STATIC,
		};
		const wrapper = shallow(
			<AppBar {...props}>
				<div>mock-children</div>
			</AppBar>,
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-app-bar-className',
			position: AppBar.PositionType.STICKY,
		};
		const wrapper = mount(
			<AppBar {...props}>
				<div>mock-children</div>
			</AppBar>
		);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().position).toEqual(props.position);
		expect(wrapper.props().children).toEqual(<div>mock-children</div>);
	})
})