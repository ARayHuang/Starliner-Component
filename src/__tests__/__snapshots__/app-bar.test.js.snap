// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app bar should render correctly 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="mock-app-bar-className"
  color="transparent"
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <div>
      mock-children
    </div>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
